README
IP-CountryFlag

The README  is  used  to  introduce the module and provide instructions on how to
install the module, any machine dependencies it may have (for example C compilers
and installed libraries) and any other information that should be provided before
the module is installed.

A  README  file  is required for CPAN modules since CPAN extracts the README file
from a module distribution so that people browsing the archive can use it to  get
an  idea  of  the  module's  uses.  It  is usually a good idea to provide version
information here so that people can decide whether fixes for the module are worth
downloading.

INSTALLATION

To install this module, run the following commands:

    perl Makefile.PL
    make
    make test
    make install

SUPPORT AND DOCUMENTATION

After  installing,   you can  find documentation for this module with the perldoc
command.

    perldoc IP::CountryFlag

You can also look for information at:

    RT, CPAN's request tracker
        http://rt.cpan.org/NoAuth/Bugs.html?Dist=IP-CountryFlag

    AnnoCPAN, Annotated CPAN documentation
        http://annocpan.org/dist/IP-CountryFlag

    CPAN Ratings
        http://cpanratings.perl.org/d/IP-CountryFlag

    Search CPAN
        http://search.cpan.org/dist/IP-CountryFlag/

LICENSE AND COPYRIGHT

Copyright (C) 2011 - 2015 Mohammad S Anwar

This program is free software; you can redistribute it and/or modify it under the
terms of the the Artistic License (2.0).You may obtain a copy of the full license
at:

L<http://www.perlfoundation.org/artistic_license_2_0>

Any  use,  modification, and distribution of the Standard or Modified Versions is
governed by this Artistic License.By using, modifying / distributing the Package,
you accept this license. Do not use, modify, or distribute the Package, if you do
not accept this license.

If your Modified Version has been derived from a Modified Version made by someone
other than you, you're nevertheless required to ensure that your Modified Version
complies with the requirements of this license.

This  license  does  not  grant you the right to use any trademark, service mark,
tradename, or logo of the Copyright Holder.

This license includes the non-exclusive, worldwide, free-of-charge patent license
to  make,  have made, use, offer to sell, sell, import and otherwise transfer the
Package with respect to any patent claims licensable by the Copyright Holder that
are  necessarily  infringed  by the  Package.  If you institute patent litigation
(including  a  cross-claim or  counterclaim)  against any party alleging that the
Package constitutes direct or contributory patent infringement,then this Artistic
License to you shall terminate on the date that such litigation is filed.

Disclaimer  of  Warranty:  THE  PACKAGE  IS  PROVIDED BY THE COPYRIGHT HOLDER AND
CONTRIBUTORS  "AS IS'  AND WITHOUT ANY EXPRESS OR IMPLIED WARRANTIES. THE IMPLIED
WARRANTIES    OF     MERCHANTABILITY,   FITNESS   FOR   A PARTICULAR  PURPOSE, OR
NON-INFRINGEMENT ARE DISCLAIMED TO THE EXTENT PERMITTED BY YOUR LOCAL LAW. UNLESS
REQUIRED BY LAW, NO COPYRIGHT HOLDER OR CONTRIBUTOR WILL BE LIABLE FOR ANY DIRECT,
INDIRECT, INCIDENTAL, OR CONSEQUENTIAL DAMAGES ARISING IN ANY WAY OUT OF THE USE
OF THE PACKAGE,EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.

Changes
Revision history for IP::CountryFlag

0.13    Mon Oct 07 15:50:00 2019
        - Moved author test scripts to xt/ subfolder.

0.12    Mon Oct 23 11:10:00 2017
        - Added test scripts (t/meta-json.t and t/meta-yml.t).

0.11    Tue Oct 10 11:00:00 2017
        - Added key 'provides' to the Makefile.PL script.

0.10    Fri Sep 22 15:40:00 2017
        - Improved method exporting.

0.09    Fri Sep 08 12:00:00 2017
        - Updated copyright year information.

0.08    Mon Sep 05 09:55:00 2016
        - Updated copyright year information.
        - Tidied up unit test script t/00-load.t.

0.07    Thu Feb 05 11:35:00 2015
        - Updated MANIFEST.SKIP file to ignore folder '_eumm'.
        - Added LICENSE file.
        - Defined $AUTHORITY in all the packages of the distribution.
        - Updated Copyright year information.

0.06    Tue Dec 16 10:53:00 2014
        - Added missing dependency 'HTTP::Tiny' to Mafefile.PL file.

0.05    Mon Dec 15 11:05:00 2014
        - Added missing dependency 'autodie' to Mafefile.PL file.
        - Minor changes to the method _save() to remove warning.

0.04    Sun Dec 14 03:00:00 2014
        - Added missing dependency 'Throwable' to Makefile.PL file.

0.03    Sat Dec 13 09:30:11 2014
        - Re-structured the code using 'Moo' i.e. downgraded from 'Mouse'.
        - Added package IP::CountryFlag::Params to support common data types.
        - Added package IP::CountryFlag::UserAgent to process api request.
        - Added package IP::CountryFlag::UserAgent::Exception to process exception.
        - Deleted Build.PL file.
        - Deleted SIGNATURE file.
        - Updated unit test with respect to the above changes.

0.02    Tue Aug 09 15:40:11 2011
        - Made Readonly v1.03 as minimum requirement.
        - Prepared the distribution using the Module::Signature v0.68.

0.01    Thu Aug 04 12:00:01 2011
        - First version, released.


lib/IP/CountryFlag.pm
package IP::CountryFlag;

$IP::CountryFlag::VERSION   = '0.13';
$IP::CountryFlag::AUTHORITY = 'cpan:MANWAR';

=head1 NAME

IP::CountryFlag - Interface to fetch country flag of an IP.

=head1 VERSION

Version 0.13

=cut

use 5.006;
use autodie;
use IP::CountryFlag::UserAgent;
use File::Spec::Functions qw(catfile);
use IP::CountryFlag::Params qw(validate);

use Moo;
use namespace::clean;
extends 'IP::CountryFlag::UserAgent';

has 'base_url' => (is => 'ro', default => sub { return 'http://api.hostip.info/flag.php' });

=head1 DESCRIPTION

A very thin wrapper for the hostip.info API to get the country flag of an IP address.

=head1 METHOD

=head2 save()

Saves the country flag in the given location for the given IP address. It returns the location
of the country flag where it has been saved.

    use strict; use warnings;
    use IP::CountryFlag;

    my $countryFlag = IP::CountryFlag->new;
    print $countryFlag->save({ ip => '12.215.42.19', path => './' });

=cut

sub save {
    my ($self, $params) = @_;

    my $fields   = { 'ip' => 1, 'path' => 1 };
    my $url      = $self->_url($fields, $params);
    my $response = $self->get($url);

    return _save($params, $response->{content});
}

#
#
# PRIVATE METHODS

sub _save {
    my ($params, $data) = @_;

    my $flag = catfile($params->{path}, $params->{ip} . ".gif");
    eval {
        open(my $FLAG, ">$flag");
        binmode($FLAG);
        print $FLAG $data;
        close $FLAG;

        return $flag;
    };

    die("ERROR: Couldn't save flag [$flag][$@].\n") if $@;
}

sub _url {
    my ($self, $fields, $params) = @_;

    validate($fields, $params);

    return sprintf("%s?ip=%s", $self->base_url, $params->{ip});
}

=head1 AUTHOR

Mohammad S Anwar, C<< <mohammad.anwar at yahoo.com> >>

=head1 REPOSITORY

L<https://github.com/manwar/IP-CountryFlag>

=head1 BUGS

Please report any bugs / feature requests to C<bug-ip-countryflag at rt.cpan.org>,
or through the web interface at L<http://rt.cpan.org/NoAuth/ReportBug.html?Queue=IP-CountryFlag>.
I will be notified & then you'll automatically be notified of progress on your bug
as I make changes.

=head1 SUPPORT

You can find documentation for this module with the perldoc command.

    perldoc IP::CountryFlag

You can also look for information at:

=over 4

=item * RT: CPAN's request tracker

L<http://rt.cpan.org/NoAuth/Bugs.html?Dist=IP-CountryFlag>

=item * AnnoCPAN: Annotated CPAN documentation

L<http://annocpan.org/dist/IP-CountryFlag>

=item * CPAN Ratings

L<http://cpanratings.perl.org/d/IP-CountryFlag>

=item * Search CPAN

L<http://search.cpan.org/dist/IP-CountryFlag/>

=back

=head1 LICENSE AND COPYRIGHT

Copyright (C) 2011 - 2017 Mohammad S Anwar.

This  program  is  free software; you can redistribute it and/or modify it under
the  terms  of the the Artistic License (2.0). You may obtain a copy of the full
license at:

L<http://www.perlfoundation.org/artistic_license_2_0>

Any  use,  modification, and distribution of the Standard or Modified Versions is
governed by this Artistic License.By using, modifying or distributing the Package,
you accept this license. Do not use, modify, or distribute the Package, if you do
not accept this license.

If your Modified Version has been derived from a Modified Version made by someone
other than you,you are nevertheless required to ensure that your Modified Version
 complies with the requirements of this license.

This  license  does  not grant you the right to use any trademark,  service mark,
tradename, or logo of the Copyright Holder.

This license includes the non-exclusive, worldwide, free-of-charge patent license
to make,  have made, use,  offer to sell, sell, import and otherwise transfer the
Package with respect to any patent claims licensable by the Copyright Holder that
are  necessarily  infringed  by  the  Package. If you institute patent litigation
(including  a  cross-claim  or  counterclaim) against any party alleging that the
Package constitutes direct or contributory patent infringement,then this Artistic
License to you shall terminate on the date that such litigation is filed.

Disclaimer  of  Warranty:  THE  PACKAGE  IS  PROVIDED BY THE COPYRIGHT HOLDER AND
CONTRIBUTORS  "AS IS'  AND WITHOUT ANY EXPRESS OR IMPLIED WARRANTIES. THE IMPLIED
WARRANTIES    OF   MERCHANTABILITY,   FITNESS   FOR   A   PARTICULAR  PURPOSE, OR
NON-INFRINGEMENT ARE DISCLAIMED TO THE EXTENT PERMITTED BY YOUR LOCAL LAW. UNLESS
REQUIRED BY LAW, NO COPYRIGHT HOLDER OR CONTRIBUTOR WILL BE LIABLE FOR ANY DIRECT,
INDIRECT, INCIDENTAL,  OR CONSEQUENTIAL DAMAGES ARISING IN ANY WAY OUT OF THE USE
OF THE PACKAGE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.

DEAFilter API itself is distributed under the terms of the Gnu GPLv3 licence.

=cut

1; # End of IP::CountryFlag


lib/IP/CountryFlag/Params.pm
package IP::CountryFlag::Params;

$IP::CountryFlag::Params::VERSION   = '0.13';
$IP::CountryFlag::Params::AUTHORITY = 'cpan:MANWAR';

=head1 NAME

IP::CountryFlag::Params - Placeholder for parameters for IP::CountryFlag.

=head1 VERSION

Version 0.13

=cut

use 5.006;
use strict; use warnings;
use Data::Validate::IP qw(is_ipv4 is_ipv6);
use parent 'Exporter';

our @EXPORT_OK = qw(validate);

sub check_ip   { die "ERROR: Received invalid IP [$_[0]]."    unless (is_ipv4($_[0]) || is_ipv6($_[0])) };
sub check_path { die "ERROR: Received invalid Path [$_[0]]."  unless (-d "$_[0]")                       };

our $FIELDS = {
    'ip'   => { check => sub { check_ip(@_)   }, type => 's' },
    'path' => { check => sub { check_path(@_) }, type => 's' },
};

sub validate {
    my ($fields, $values) = @_;

    die "ERROR: Missing params list." unless (defined $values);

    die "ERROR: Parameters have to be hash ref" unless (ref($values) eq 'HASH');

    foreach my $field (sort keys %{$fields}) {
        die "ERROR: Received invalid param: $field"
            unless (exists $FIELDS->{$field});

        die "ERROR: Missing mandatory param: $field"
            if ($fields->{$field} && !exists $values->{$field});

        die "ERROR: Received undefined mandatory param: $field"
            if ($fields->{$field} && !defined $values->{$field});

	$FIELDS->{$field}->{check}->($values->{$field})
            if defined $values->{$field};
    }
}

=head1 AUTHOR

Mohammad S Anwar, C<< <mohammad.anwar at yahoo.com> >>

=head1 REPOSITORY

L<https://github.com/manwar/IP-CountryFlag>

=head1 BUGS

Please report any bugs / feature requests to C<bug-ip-countryflag at rt.cpan.org>,
or through the web interface at L<http://rt.cpan.org/NoAuth/ReportBug.html?Queue=IP-CountryFlag>.
I will be notified & then you'll automatically be notified of progress on your bug
as I make changes.

=head1 SUPPORT

You can find documentation for this module with the perldoc command.

    perldoc IP::CountryFlag::Params

You can also look for information at:

=over 4

=item * RT: CPAN's request tracker

L<http://rt.cpan.org/NoAuth/Bugs.html?Dist=IP-CountryFlag>

=item * AnnoCPAN: Annotated CPAN documentation

L<http://annocpan.org/dist/IP-CountryFlag>

=item * CPAN Ratings

L<http://cpanratings.perl.org/d/IP-CountryFlag>

=item * Search CPAN

L<http://search.cpan.org/dist/IP-CountryFlag/>

=back

=head1 LICENSE AND COPYRIGHT

Copyright (C) 2011 - 2017 Mohammad S Anwar.

This  program  is  free software; you can redistribute it and/or modify it under
the  terms  of the the Artistic License (2.0). You may obtain a copy of the full
license at:

L<http://www.perlfoundation.org/artistic_license_2_0>

Any  use,  modification, and distribution of the Standard or Modified Versions is
governed by this Artistic License.By using, modifying or distributing the Package,
you accept this license. Do not use, modify, or distribute the Package, if you do
not accept this license.

If your Modified Version has been derived from a Modified Version made by someone
other than you,you are nevertheless required to ensure that your Modified Version
 complies with the requirements of this license.

This  license  does  not grant you the right to use any trademark,  service mark,
tradename, or logo of the Copyright Holder.

This license includes the non-exclusive, worldwide, free-of-charge patent license
to make,  have made, use,  offer to sell, sell, import and otherwise transfer the
Package with respect to any patent claims licensable by the Copyright Holder that
are  necessarily  infringed  by  the  Package. If you institute patent litigation
(including  a  cross-claim  or  counterclaim) against any party alleging that the
Package constitutes direct or contributory patent infringement,then this Artistic
License to you shall terminate on the date that such litigation is filed.

Disclaimer  of  Warranty:  THE  PACKAGE  IS  PROVIDED BY THE COPYRIGHT HOLDER AND
CONTRIBUTORS  "AS IS'  AND WITHOUT ANY EXPRESS OR IMPLIED WARRANTIES. THE IMPLIED
WARRANTIES    OF   MERCHANTABILITY,   FITNESS   FOR   A   PARTICULAR  PURPOSE, OR
NON-INFRINGEMENT ARE DISCLAIMED TO THE EXTENT PERMITTED BY YOUR LOCAL LAW. UNLESS
REQUIRED BY LAW, NO COPYRIGHT HOLDER OR CONTRIBUTOR WILL BE LIABLE FOR ANY DIRECT,
INDIRECT, INCIDENTAL,  OR CONSEQUENTIAL DAMAGES ARISING IN ANY WAY OUT OF THE USE
OF THE PACKAGE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.

DEAFilter API itself is distributed under the terms of the Gnu GPLv3 licence.

=cut

1; # End of IP::CountryFlag::Params


lib/IP/CountryFlag/UserAgent.pm
package IP::CountryFlag::UserAgent;

$IP::CountryFlag::UserAgent::VERSION   = '0.13';
$IP::CountryFlag::UserAgent::AUTHORITY = 'cpan:MANWAR';

=head1 NAME

IP::CountryFlag::UserAgent - Interface to user agent for hostip.info API.

=head1 VERSION

Version 0.13

=cut

use 5.006;

use HTTP::Tiny;
use IP::CountryFlag::UserAgent::Exception;

use Moo;
use namespace::clean;

has 'ua' => (is => 'rw', default  => sub { HTTP::Tiny->new(agent => "IP/CountryFlag/v1"); } );

=head1 METHODS

=head2 get($url)

The method get() expects one parameter i.e. URL and returns the standard response.
On error throws exception of type L<WebService::Wikimapia::UserAgent::Exception>.

=cut

sub get {
    my ($self, $url) = @_;

    my $ua       = $self->ua;
    my $response = $ua->request('GET', $url);
    my @caller   = caller(1);
    @caller      = caller(2) if $caller[3] eq '(eval)';

    unless ($response->{success}) {
	IP::CountryFlag::UserAgent::Exception->throw({
            method      => $caller[3],
            message     => "request to API failed",
            code        => $response->{status},
            reason      => $response->{reason},
            filename    => $caller[1],
            line_number => $caller[2] });
    }

    return $response;
}

=head1 AUTHOR

Mohammad S Anwar, C<< <mohammad.anwar at yahoo.com> >>

=head1 REPOSITORY

L<https://github.com/manwar/IP-CountryFlag>

=head1 BUGS

Please report any bugs / feature requests to C<bug-ip-countryflag at rt.cpan.org>,
or through the web interface at L<http://rt.cpan.org/NoAuth/ReportBug.html?Queue=IP-CountryFlag>.
I will be notified & then you'll automatically be notified of progress on your bug
as I make changes.

=head1 SUPPORT

You can find documentation for this module with the perldoc command.

    perldoc IP::CountryFlag::UserAgent

You can also look for information at:

=over 4

=item * RT: CPAN's request tracker

L<http://rt.cpan.org/NoAuth/Bugs.html?Dist=IP-CountryFlag>

=item * AnnoCPAN: Annotated CPAN documentation

L<http://annocpan.org/dist/IP-CountryFlag>

=item * CPAN Ratings

L<http://cpanratings.perl.org/d/IP-CountryFlag>

=item * Search CPAN

L<http://search.cpan.org/dist/IP-CountryFlag/>

=back

=head1 LICENSE AND COPYRIGHT

Copyright (C) 2011 - 2017 Mohammad S Anwar.

This  program  is  free software; you can redistribute it and/or modify it under
the  terms  of the the Artistic License (2.0). You may obtain a copy of the full
license at:

L<http://www.perlfoundation.org/artistic_license_2_0>

Any  use,  modification, and distribution of the Standard or Modified Versions is
governed by this Artistic License.By using, modifying or distributing the Package,
you accept this license. Do not use, modify, or distribute the Package, if you do
not accept this license.

If your Modified Version has been derived from a Modified Version made by someone
other than you,you are nevertheless required to ensure that your Modified Version
 complies with the requirements of this license.

This  license  does  not grant you the right to use any trademark,  service mark,
tradename, or logo of the Copyright Holder.

This license includes the non-exclusive, worldwide, free-of-charge patent license
to make,  have made, use,  offer to sell, sell, import and otherwise transfer the
Package with respect to any patent claims licensable by the Copyright Holder that
are  necessarily  infringed  by  the  Package. If you institute patent litigation
(including  a  cross-claim  or  counterclaim) against any party alleging that the
Package constitutes direct or contributory patent infringement,then this Artistic
License to you shall terminate on the date that such litigation is filed.

Disclaimer  of  Warranty:  THE  PACKAGE  IS  PROVIDED BY THE COPYRIGHT HOLDER AND
CONTRIBUTORS  "AS IS'  AND WITHOUT ANY EXPRESS OR IMPLIED WARRANTIES. THE IMPLIED
WARRANTIES    OF   MERCHANTABILITY,   FITNESS   FOR   A   PARTICULAR  PURPOSE, OR
NON-INFRINGEMENT ARE DISCLAIMED TO THE EXTENT PERMITTED BY YOUR LOCAL LAW. UNLESS
REQUIRED BY LAW, NO COPYRIGHT HOLDER OR CONTRIBUTOR WILL BE LIABLE FOR ANY DIRECT,
INDIRECT, INCIDENTAL,  OR CONSEQUENTIAL DAMAGES ARISING IN ANY WAY OUT OF THE USE
OF THE PACKAGE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.

DEAFilter API itself is distributed under the terms of the Gnu GPLv3 licence.

=cut

1; # End of IP::CountryFlag::UserAgent


lib/IP/CountryFlag/UserAgent/Exception.pm
package IP::CountryFlag::UserAgent::Exception;

$IP::CountryFlag::UserAgent::Exception::VERSION   = '0.13';
$IP::CountryFlag::UserAgent::Exception::AUTHORITY = 'cpan:MANWAR';

=head1 NAME

IP::CountryFlag::UserAgent::Exception - Exception handler for the module IP::CountryFlag.

=head1 VERSION

Version 0.13

=cut

use 5.006;

use Moo;
use namespace::clean;
with 'Throwable';

use overload q{""} => 'as_string', fallback => 1;

has method      => (is => 'ro');
has message     => (is => 'ro');
has code        => (is => 'ro');
has reason      => (is => 'ro');
has filename    => (is => 'ro');
has line_number => (is => 'ro');

sub as_string {
    my ($self) = @_;

    return sprintf("%s(): %s (status: %s) file %s on line %d\n",
                   $self->method, $self->message, $self->code,
                   $self->filename, $self->line_number);
}

=head1 AUTHOR

Mohammad S Anwar, C<< <mohammad.anwar at yahoo.com> >>

=head1 REPOSITORY

L<https://github.com/manwar/IP-CountryFlag>

=head1 BUGS

Please report any bugs / feature requests to C<bug-ip-countryflag at rt.cpan.org>,
or through the web interface at L<http://rt.cpan.org/NoAuth/ReportBug.html?Queue=IP-CountryFlag>.
I will be notified & then you'll automatically be notified of progress on your bug
as I make changes.

=head1 SUPPORT

You can find documentation for this module with the perldoc command.

    perldoc IP::CountryFlag::UserAgent::Exception

You can also look for information at:

=over 4

=item * RT: CPAN's request tracker

L<http://rt.cpan.org/NoAuth/Bugs.html?Dist=IP-CountryFlag>

=item * AnnoCPAN: Annotated CPAN documentation

L<http://annocpan.org/dist/IP-CountryFlag>

=item * CPAN Ratings

L<http://cpanratings.perl.org/d/IP-CountryFlag>

=item * Search CPAN

L<http://search.cpan.org/dist/IP-CountryFlag/>

=back

=head1 LICENSE AND COPYRIGHT

Copyright (C) 2011 - 2017 Mohammad S Anwar.

This  program  is  free software; you can redistribute it and/or modify it under
the  terms  of the the Artistic License (2.0). You may obtain a copy of the full
license at:

L<http://www.perlfoundation.org/artistic_license_2_0>

Any  use,  modification, and distribution of the Standard or Modified Versions is
governed by this Artistic License.By using, modifying or distributing the Package,
you accept this license. Do not use, modify, or distribute the Package, if you do
not accept this license.

If your Modified Version has been derived from a Modified Version made by someone
other than you,you are nevertheless required to ensure that your Modified Version
 complies with the requirements of this license.

This  license  does  not grant you the right to use any trademark,  service mark,
tradename, or logo of the Copyright Holder.

This license includes the non-exclusive, worldwide, free-of-charge patent license
to make,  have made, use,  offer to sell, sell, import and otherwise transfer the
Package with respect to any patent claims licensable by the Copyright Holder that
are  necessarily  infringed  by  the  Package. If you institute patent litigation
(including  a  cross-claim  or  counterclaim) against any party alleging that the
Package constitutes direct or contributory patent infringement,then this Artistic
License to you shall terminate on the date that such litigation is filed.

Disclaimer  of  Warranty:  THE  PACKAGE  IS  PROVIDED BY THE COPYRIGHT HOLDER AND
CONTRIBUTORS  "AS IS'  AND WITHOUT ANY EXPRESS OR IMPLIED WARRANTIES. THE IMPLIED
WARRANTIES    OF   MERCHANTABILITY,   FITNESS   FOR   A   PARTICULAR  PURPOSE, OR
NON-INFRINGEMENT ARE DISCLAIMED TO THE EXTENT PERMITTED BY YOUR LOCAL LAW. UNLESS
REQUIRED BY LAW, NO COPYRIGHT HOLDER OR CONTRIBUTOR WILL BE LIABLE FOR ANY DIRECT,
INDIRECT, INCIDENTAL,  OR CONSEQUENTIAL DAMAGES ARISING IN ANY WAY OUT OF THE USE
OF THE PACKAGE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.

DEAFilter API itself is distributed under the terms of the Gnu GPLv3 licence.

=cut

1; # End of IP::CountryFlag::UserAgent::Exception


Makefile.PL
#!/usr/bin/perl

use 5.006;
use strict;
use warnings FATAL => 'all';
use ExtUtils::MakeMaker;

WriteMakefile(
    NAME               => 'IP::CountryFlag',
    AUTHOR             => q{Mohammad S Anwar <mohammad.anwar@yahoo.com>},
    VERSION_FROM       => 'lib/IP/CountryFlag.pm',
    ABSTRACT_FROM      => 'lib/IP/CountryFlag.pm',
    LICENSE            => 'artistic_2',
    MIN_PERL_VERSION   => 5.006,
    CONFIGURE_REQUIRES => {
        'ExtUtils::MakeMaker' => 0,
    },
    BUILD_REQUIRES => {
        'Test::More' => 0,
    },
    PREREQ_PM => {
        'autodie'               => '2.06',
        'Moo'                   => '2.000000',
        'namespace::clean'      => '0.24',
        'Throwable'             => 0,
        'HTTP::Tiny'            => 0,
        'Data::Validate::IP'    => '0.10',
        'File::Spec::Functions' => '3.30',
    },
    dist  => { COMPRESS => 'gzip -9f', SUFFIX => 'gz', },
    clean => { FILES => 'IP-CountryFlag-*' },
    (eval { ExtUtils::MakeMaker->VERSION(6.46) } ? (META_MERGE => {
        'meta-spec' => { version => 2 },
        provides    => {
            'IP::CountryFlag'                       => { file => 'lib/IP/CountryFlag.pm',                     version => '0.13' },
            'IP::CountryFlag::Params'               => { file => 'lib/IP/CountryFlag/Params.pm',              version => '0.13' },
            'IP::CountryFlag::UserAgent'            => { file => 'lib/IP/CountryFlag/UserAgent.pm',           version => '0.13' },
            'IP::CountryFlag::UserAgent::Exception' => { file => 'lib/IP/CountryFlag/UserAgent/Exception.pm', version => '0.13' },
        },
        resources => {
            repository => {
                type => 'git',
                url  => 'https://github.com/manwar/IP-CountryFlag.git',
                web  => 'https://github.com/manwar/IP-CountryFlag',
            },
        }})
     : ()
    ),
);
